engine/src/lib.rs

```diff
@@ -69,20 +62,66 @@ impl ServalEngine {
     /// Run the passed-in WASM executable on the given input bytes.
     pub fn execute(
         &mut self,
+        // WebAssembly binary to execute
         binary: &[u8],
+        // Data to pass to WebAssembly as stdin
         input: &[u8],
+        // List of elevated permissions for this binary
+        permissions: &[Permission],
     ) -> Result<WasmResult, ServalEngineError> {
         let stdout = WritePipe::new_in_memory();
         let stderr = WritePipe::new_in_memory();
 
+        // Link the experimental HTTP support
+        let allowed_http_hosts = if permissions.contains(&Permission::AllHttpHosts) {
+            // todo: unclear whether we should actually support a wildcard like this
```

Comment:
yeah I'd like to slow-walk this wildcard idea a little bit. i can see applications that would need it but let's make sure the restricted case is rock-solid first

Reply:
That's my thinking as well, although it's preferable to have it while we're in wild hacking mode.